test/functional/executor.py

```diff
@@ -103,6 +124,26 @@ def is_gold_and_output_equal(self):
             print('PASS: {} EQUALS {}'.format(self.output_filename, self.golden_filename))
             return True
 
+    def is_standard_output_equal(self):
+        """Compares the standard output with the reference file.
+
+        Returns:
+        True if the standard output and the reference file are equal, false otherwise.
+        """
+
+        line_number = 1
+        with open(self.stdout_filename) as stdout_ref, open(self.stdout_output) as stdout_out:
+            for stdout_line, stdref_line in itertools.zip_longest(stdout_out, stdout_ref):
```

Comment:
How does this handle difference in paths? Is this regex responsible for that?

Reply:
Yes, the regex removes the path leaving just the file name. E.g. '/some/path/to/the/filename' will be reduced to 'filename'. Still it is better not to keep the whole path in the reference files as you noticed above. Anyway, this shortening of paths will be left here.

Reply:
Why not `os.path.basename()` then?

Reply:
The truth is that the algorithm doesn't know when it is dealing with the path name. It is just searching for some text that looks like a path (just `/some/text/between/slashes/is/treated/as/a/path/`). When found, the path is removed. The whole file is treated as a blob of text. To use `os.path.basename()` we would have to find something that looks like a path and a filename and then use the function.

Reply:
Just added also handling Windows paths. As mentioned before we cannot use `os.path.basename()` as the standard output of the application differs heavily fro case to case and we never know when we deal with a path. So the output text is stripped from anything that resembles a path name.

Reply:
But we know how vcdmaker outputs stuff. If stdout line begins with "Parsed " then we know the next thing is a path and we could use that knowledge. If you check if both stdout lines begin with "Parsed " then you can just continue. Who said you have to blindly compare stdouts line by line, we know our stdout, right?

Reply:
Unfortunatelly there are more different contexts (standard output and errors). In the future there might be even more.

Current examples.

Basic one:
```
Parsed /some/path/here/test_003.txt
```

Merging error:
```
[Error 0002]: Inconsistent signal: A.Val. Types: wire / wire. Sizes: 8 / 9. Sources: /some/path/here/test_007.txt and /other/path/here/test_007.txt. At line 2.
```

Merging source description:
```
U{../some/path/here/vcdMaker.xml},71050601,us,System1,Counter,system1_time1.txt
```

I even thoght about putting paths and file names inside some markers but I do not the idea either. What do you think about it?

Reply:
I just tried to prepared regexes for all the cases to filter out path names. The regexes get so complicated that I'd rather stick with the current solution.

Reply:
```
for expected_line in expected_output and actual_line in actual_output do
    if actual_line begins with "Parsed " and expected_line begins with "Parsed " then
        continue loop
    else
        fail test

    if actual_line != expected_line then
        fail test
```
Isn't this sufficient?

Reply:
As we discussed there were more contexts in which the path and the file name appeared. The current solution seems just reasonable.